lib/symboldatabase.cpp

```diff
@@ -283,8 +292,13 @@ void SymbolDatabase::createSymbolDatabaseFindAllScopes()
                 tok = tok->tokAt(3);
 
             // skip over qualification
-            while (Token::Match(tok, "%type% ::"))
-                tok = tok->tokAt(2);
+            while (Token::Match(tok, "%type% ::") ||
```

Comment:
hmm.. as far as I see we have 3 identical loops here. and the condition `(Token::Match(tok, "%type% <") && tok->linkAt(1) && tok->linkAt(1)->strAt(1) == "::")` is repeated over and over..